### PR TITLE
[QUIZ-582] subtask update quizzes UI with skip button

### PR DESCRIPTION
**Pull request by v-i-s-h-n-u-ps**
This PR adds the ability to skip questions that are marked as skippable.

INDEX_KEY: `key_wJSdZSiesX5hiVLt`

A couple of questions in the `coffee-quiz` are marked skippable.

https://app.constructor.io/dashboard/quizzes/edit/coffee-quiz?back=%2Fdashboard%2Fquizzes%3Findex_key%3Dkey_wJSdZSiesX5hiVLt

### CODE REVIEW

File: src/hooks/useQuizEvents/useQuizSkipClick.ts
Comment (mocca102) [inline src/hooks/useQuizEvents/useQuizSkipClick.ts:8]: I believe you meant `useQuizSkipClick` here

File: src/hooks/useQuizEvents/useQuizSkipClick.ts
Comment (mocca102) [inline src/hooks/useQuizEvents/useQuizSkipClick.ts:14]: I believe you meant `quizSkipClickHandler`

Comment (v-i-s-h-n-u-ps) [reply src/hooks/useQuizEvents/useQuizSkipClick.ts]: [39da1d4](https://github.com/Constructor-io/constructorio-ui-quizzes/pull/132/commits/39da1d4e2a01f97440c535a87b5777047d6a8499)

File: src/hooks/useQuizEvents/useQuizSkipClick.ts
Comment (v-i-s-h-n-u-ps) [reply src/hooks/useQuizEvents/useQuizSkipClick.ts]: [39da1d4](https://github.com/Constructor-io/constructorio-ui-quizzes/pull/132/commits/39da1d4e2a01f97440c535a87b5777047d6a8499)

**Review by mocca102 — APPROVED** [review]
LGTM! lets get a second review as well

File: src/components/CioQuiz/quizLocalReducer.ts
Comment (mocca102) [inline src/components/CioQuiz/quizLocalReducer.ts]: nice refactor

File: src/components/CioQuiz/quizLocalReducer.ts
Comment (esezen) [inline src/components/CioQuiz/quizLocalReducer.ts:100]: The previous line has optional chaining, does that mean `currentAnswerInput` could be null/undefined? If yes, is it possible for currentAnswerInput.type to error out? I just realized it's the same in `Next` so maybe it's a question for @mocca102 as well

File: src/components/CioQuiz/quizLocalReducer.ts
Comment (esezen) [inline src/components/CioQuiz/quizLocalReducer.ts:115]: I'm not super familiar with this file's logic but should we be writing to `prevAnswerInputs` here?

File: src/components/CioQuiz/quizLocalReducer.ts
Comment (esezen) [inline src/components/CioQuiz/quizLocalReducer.ts:116]: Do we need to write `answerInputs` here?

File: src/stories/Quiz/tests/mocks.ts
Comment (esezen) [inline src/stories/Quiz/tests/mocks.ts:210]: Should this be `cio-question-skip-button` or does it not matter?

**Review by esezen — COMMENTED** [review]
Thanks for working on this, this is looking great!

File: src/components/CioQuiz/quizLocalReducer.ts
Comment (mocca102) [reply src/components/CioQuiz/quizLocalReducer.ts:100]: The optional chaining actually should be removed.

Comment (mocca102) [reply src/components/CioQuiz/quizLocalReducer.ts:100]: It's not doing anything but it's throwing whomever reading the code off

File: src/stories/Quiz/tests/mocks.ts
Comment (v-i-s-h-n-u-ps) [reply src/stories/Quiz/tests/mocks.ts]: [f18aa78](https://github.com/Constructor-io/constructorio-ui-quizzes/pull/132/commits/f18aa78821419475ac9b701bcd44f4f165819fca)

File: src/components/CioQuiz/quizLocalReducer.ts
Comment (v-i-s-h-n-u-ps) [reply src/components/CioQuiz/quizLocalReducer.ts:100]: Removed [f18aa78](https://github.com/Constructor-io/constructorio-ui-quizzes/pull/132/commits/f18aa78821419475ac9b701bcd44f4f165819fca)

File: src/components/CioQuiz/quizLocalReducer.ts
Comment (esezen) [reply src/components/CioQuiz/quizLocalReducer.ts:115]: @mocca102 thoughts?

File: src/components/CioQuiz/quizLocalReducer.ts
Comment (esezen) [reply src/components/CioQuiz/quizLocalReducer.ts:116]: @mocca102 thoughts?

File: src/components/CioQuiz/quizLocalReducer.ts
Comment (mocca102) [reply src/components/CioQuiz/quizLocalReducer.ts:115]: What do you mean by? @esezen 

>  writing to prevAnswerInputs here?

**Review by esezen — APPROVED** [review]
LGTM!

**Pull request merged**
